spec/lib/serialization_core_spec.rb

```diff
@@ -1,82 +1,210 @@
+        expect(ids_hash).to be_empty
       end
+    end
 
-      result = MovieSerializer.send(:ids_hash, [], 'movie')
-      expect(result).to be_empty
+    context 'when ids is just a single id' do
+      let(:ids) { '1' }
+
+      it 'returns correct hash' do
+        expect(ids_hash[:id]).to eq '1'
+        expect(ids_hash[:type]).to eq :movie
+      end
+    end
+
+    context 'when ids is nil' do
+      let(:ids) { nil }
+
+      it 'returns correct hash' do
+        expect(ids_hash).to be_nil
+      end
+    end
+  end
+
+  describe '#self.ids_hash_from_record_and_relationship' do
+    context 'when an association is not polymorphic' do
+      subject(:ids_hash) { MovieSerializer.ids_hash_from_record_and_relationship(movie, relationship) }
+
+      let(:relationship) { { name: :actors, relationship_type: :has_many, object_method_name: :actors, id_method_name: :actor_ids, record_type: :actor } }
+
+      it 'returns the correct hash' do
+        expect(ids_hash.count).to eq 3
+        expect(ids_hash).to include({ id: '1', type: :actor }, { id: '2', type: :actor }, { id: '3', type: :actor })
+      end
+    end
+
+    context 'when an association is polymorphic' do
+      subject(:ids_hash) { GroupSerializer.ids_hash_from_record_and_relationship(group, relationship) }
+
+      let(:relationship) { { name: :groupees, relationship_type: :has_many, object_method_name: :groupees, polymorphic: {} } }
+
+      it 'returns the correct hash' do
+        expect(ids_hash.count).to eq 2
+        expect(ids_hash).to include({ id: '1', type: :person }, { id: '2', type: :group })
+      end
     end
+  end
+
+  describe '#self.attributes_hash' do
+    subject(:attributes_hash) { MovieSerializer.attributes_hash(movie) }
+    let(:attributes_to_serialize) { MovieSerializer.attributes_to_serialize }
 
-    it 'returns correct hash when attributes_hash is called' do
-      attributes_hash = MovieSerializer.send(:attributes_hash, movie)
-      attribute_names = attributes_hash.keys.sort
-      expect(attribute_names).to eq MovieSerializer.attributes_to_serialize.keys.sort
-      MovieSerializer.attributes_to_serialize.each do |key, method_name|
-        value = attributes_hash[key]
-        expect(value).to eq movie.send(method_name)
+    context 'when attribute without block' do
+      it 'returns correct hash' do
+        expect(attributes_hash.keys).to eq MovieSerializer.attributes_to_serialize.keys
+        attributes_to_serialize.each do |key, method_name|
+          expect(attributes_hash[key]).to eq movie.send(method_name)
+        end
       end
     end
 
-    it 'returns the correct empty result when relationships_hash is called' do
-      movie.actor_ids = []
-      movie.owner_id = nil
-      relationships_hash = MovieSerializer.send(:relationships_hash, movie)
-      expect(relationships_hash[:actors][:data]).to eq([])
-      expect(relationships_hash[:owner][:data]).to eq(nil)
+    context 'when attribute without block' do
```

Comment:
Should it be when attribute with block?

```ruby
context 'when attribute with block' do
 #...
end 
``` 
As you are passing a block to `MovieSerializer.attribute` ?